File: server/src/main/com/broll/gainea/server/core/cards/impl/direct/C_Phoenix.java
package com.broll.gainea.server.core.cards.impl.direct;

import com.broll.gainea.server.core.cards.DirectlyPlayedCard;
import com.broll.gainea.server.core.objects.monster.Monster;
import com.broll.gainea.server.core.objects.monster.MonsterMotion;

public class C_Phoenix extends DirectlyPlayedCard {

    public C_Phoenix() {
        super(65, "Beschworener Phönix", "Beschwört einen Phönix (2/2), dieser kann 2 Felder pro Zug bewegt werden aber kann nicht angreifen.");
        setDrawChance(0.8f);
    }

    @Override
    protected void play() {
        Monster monster = new Monster();
        monster.setName("Phönix");
        monster.getAttacksPerTurn().setValue(0);
        monster.getMovesPerTurn().setValue(2);
        monster.setHealth(2);
        monster.setPower(2);
        monster.setIcon(124);
        monster.setMotion(MonsterMotion.AIRBORNE);
        monster.setOwner(owner);
        placeUnitHandler.placeUnit(owner, monster, owner.getControlledLocations(), "Ort der Beschwörung wählen");
    }

}

File: server/src/main/com/broll/gainea/server/core/cards/impl/direct/C_Phoenix.kt
package com.broll.gainea.server.core.cards.impl.direct

import com.broll.gainea.server.core.cards.DirectlyPlayedCard
import com.broll.gainea.server.core.cards.EffectType.SUMMON
import com.broll.gainea.server.core.objects.monster.Monster
import com.broll.gainea.server.core.objects.monster.MonsterMotion

class C_Phoenix : DirectlyPlayedCard(
    65, SUMMON,
    "Beschworener Phönix",
    "Beschwört einen Phönix (2/2), dieser kann 2 Felder pro Zug bewegt werden aber kann nicht angreifen."
) {
    init {
        drawChance = 0.8f
    }

    override fun play() {
        val monster = Monster(owner)
        monster.name = "Phönix"
        monster.attacksPerTurn.value = 0
        monster.movesPerTurn.value = 2
        monster.setHealth(2)
        monster.setPower(2)
        monster.icon = 124
        monster.motion = MonsterMotion.AIRBORNE
        placeUnitHandler.placeUnit(owner, monster, owner.controlledLocations.toList())
    }
}
